src/plugins/BrowserStorageKeyStore.ts

```diff
@@ -0,0 +1,126 @@
+import { getKeyMetadata } from "../helpers/getKeyMetadata";
+import { EncryptedKey, KeyMetadata, KeyStore } from "../types";
+import {
+  BrowserStorageConfigParams,
+  BrowserStorageFacade,
+} from "./BrowserStorageFacade";
+
+/**
+ * KeyStore for Chrome and Firefox browser storage API:
+ * https://developer.chrome.com/docs/extensions/reference/storage/.
+ * Once instantiated and configured, pass it to the `KeyManager` contructor to
+ * handle the storage of encrypted keys.
+ * ```js
+ * const browserKeyStore = new KeyManagerPlugins.BrowserStorageKeyStore();
+ * browserKeyStore.configure({ storage: chrome.storage.local });
+ * const keyManager = new KeyManager({
+ *     keyStore: browserKeyStore
+ *  });
+ * ```
+ */
+export class BrowserStorageKeyStore implements KeyStore {
+  public name: string;
+  private keyStore: BrowserStorageFacade;
+
+  constructor() {
+    this.name = "BrowserStorageKeyStore";
+    this.keyStore = new BrowserStorageFacade();
+  }
+
+  /**
+   * The configuration is where the storage engine is set up and configured.
+   * It must follow the Storage interface :
+   * https://developer.chrome.com/docs/extensions/reference/storage/).
+   * This is mostly for use with Chrome and Firefox storage in addition to
+   * libraries that shim this (for ex: webextension-polyfill)
+   * @param {BrowserStorageConfigParams} params A configuration object.
+   * @param {Storage} params.storage The Storage instance. Required.
+   * @param {string} [params.prefix] The prefix for the names in the storage.
+   * @return {Promise}
+   */
+  public configure(params: BrowserStorageConfigParams) {
+    try {
+      this.keyStore.configure(params);
+      return Promise.resolve();
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+
+  public async storeKeys(keys: EncryptedKey[]) {
+    // We can't store keys if they're already there
+    const invalidKeys: EncryptedKey[] = [];
```

Comment:
`invalidKeys` sounds like there is something wrong with them. Maybe `usedKeys` or something like that would be a better name? :thinking: